beacon-chain/rpc/eth/validator/handlers.go

```diff
@@ -52,61 +54,133 @@ func (s *Server) GetAggregateAttestation(w http.ResponseWriter, r *http.Request)
 		return
 	}
 
+	match := s.aggregateAttestation(w, primitives.Slot(slot), "", attDataRoot)
+	if match == nil {
+		return
+	}
+
+	matchedAtt, ok := match.(*ethpbalpha.Attestation)
+	if !ok {
+		httputil.HandleError(w, "Match is not of type Attestation", http.StatusInternalServerError)
+		return
+	}
+	att := structs.AttFromConsensus(matchedAtt)
+	data, err := json.Marshal(att)
+	if err != nil {
+		httputil.HandleError(w, "Could not marshal attestation: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+	httputil.WriteJson(w, &structs.AggregateAttestationResponse{Data: data})
+}
+
+// GetAggregateAttestationV2 aggregates all attestations matching the given attestation data root and slot, returning the aggregated result.
+func (s *Server) GetAggregateAttestationV2(w http.ResponseWriter, r *http.Request) {
+	_, span := trace.StartSpan(r.Context(), "validator.GetAggregateAttestationV2")
+	defer span.End()
+
+	_, attDataRoot, ok := shared.HexFromQuery(w, r, "attestation_data_root", fieldparams.RootLength, true)
+	if !ok {
+		return
+	}
+	_, slot, ok := shared.UintFromQuery(w, r, "slot", true)
+	if !ok {
+		return
+	}
+	_, index, ok := shared.UintFromQuery(w, r, "committee_index", true)
+	if !ok {
+		return
+	}
```

Comment:
These should return a 400 / bad request since they are required fields.

Reply:
They already do, within `shared.UintFromQuery` we have `ValidateUint` which handles the http error, same thing applies to `shared.HexFromQuery`